cmd/runsim/integrations/github/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/cosmos/tools/lib/runsimaws"
+	"github.com/cosmos/tools/lib/runsimgh"
+)
+
+const (
+	// Security token IDs
+	ssmGhAppTokenId  = "github-sim-app-key"
+	ssmCircleTokenId = "circle-token-sim"
+
+	// Github app parameters
+	startSimCmd      = "Start sim"
+	startSimCmdDev	 = "Start sim dev"
+	ghCheckName      = "Long sim"
+	appIntegrationId  = "30867"
+	appInstallationId = "997580"
+
+	// The value to use in the conclusion field of a github check in case of failure.
+	ghConclusionFail = "failure"
+
+	// DynamoDB attribute and table names
+	awsRegion = "us-east-1"
+	simStateTable = "SimulationState" // name of the dynamodb table where details from running simulations are stored
+	primaryKey    = "IntegrationType" // primary partition key used by the sim state table
+)
+
+type CircleApiPayload struct {
+	Revision        string `json:"revision"`
+	BuildParameters struct {
+		CommitHash      string `json:"gaia-commit-hash"`
+		IntegrationType string `json:"integration-type"`
+	} `json:"parameters"`
+}
+
+type GithubEventPayload struct {
+	Issue struct {
+		Number int `json:"number"`
+		Pr     struct {
+			Url string `json:"url,omitempty"`
+		} `json:"pull_request,omitempty"`
+	} `json:"issue"`
+
+	Comment struct {
+		Body string `json:"body"`
+	} `json:"comment"`
+
+	Repo struct {
+		Name  string `json:"name"`
+		Owner struct {
+			Login string `json:"login"`
+		} `json:"owner"`
+	} `json:"repository"`
+}
+
+func handler(request events.APIGatewayProxyRequest) (response events.APIGatewayProxyResponse, err error) {
+	ddb := new(runsimaws.DdbTable)
+	ddb.Config(awsRegion, primaryKey, simStateTable)
+
+	github := new(runsimgh.Integration)
+	// Try to retrieve sim state from database. Success indicates another sim is still running
+	if _ = ddb.GetState("GitHub", github); github.IntegrationType != nil {
+		// TODO: send this response as a PR comment or some other way to notify the user
+		return buildProxyResponse(200, "INFO: another sim is already in progress"), err
+	}
+
+	var ghEvent GithubEventPayload
+	if err = json.Unmarshal([]byte(request.Body), &ghEvent); err != nil {
+		log.Printf("INFO: github request: %+v", request.Body)
+		return buildProxyResponse(500, "ERROR: unmarshal github request"), err
```

Comment:
we should wrap this error

Reply:
What's the benefit or wrapping here? This error can't be handled as the return here ends execution.